ui/src/api/files/__tests__/utils.test.ts

```diff
@@ -1,45 +1,71 @@
 import { describe, expect, test } from "vitest";
 import {
   forceLeadingSlash,
+  getFilenameFromPath,
+  getParentFromPath,
   removeLeadingSlash,
   removeTrailingSlash,
   sortByName,
   sortFoldersFirst,
 } from "../utils";
 
-import { NodeSchemaType } from "../schema/node";
+import { BaseFileSchemaType } from "../schema";
 
-const itemTemplate: NodeSchemaType = {
+const itemTemplate: BaseFileSchemaType = {
   createdAt: "2023-03-13T13:39:05.832664Z",
   updatedAt: "2023-03-13T13:39:06.118436Z",
-  name: "demo-workflow",
   path: "/demo-workflow",
-  parent: "/",
   type: "directory",
-  attributes: [],
-  readOnly: true,
-  expandedType: "git",
-  mimeType: "text/plain; charset=utf-8",
 };
 
+describe("getFilenameFromPath()", () => {
+  test("file at root level", () => {
+    expect(getFilenameFromPath("/action.yaml")).toBe("action.yaml");
+  });
+  test("file in directory", () => {
+    expect(getFilenameFromPath("/foobar/nested.yaml")).toBe("nested.yaml");
+  });
+  test("directory", () => {
+    expect(getFilenameFromPath("/examples")).toBe("examples");
+  });
+  test("nested directory", () => {
+    expect(getFilenameFromPath("/examples/nested")).toBe("nested");
+  });
+});
+
+describe("getParentFromPath()", () => {
+  test("file at root level", () => {
+    expect(getParentFromPath("/action.yaml")).toBe("/");
+  });
+  test("file in directory", () => {
+    expect(getParentFromPath("/foobar/nested.yaml")).toBe("/foobar");
+  });
+  test("directory at root level", () => {
+    expect(getParentFromPath("/examples")).toBe("/");
+  });
+  test("nested directory", () => {
+    expect(getParentFromPath("/examples/nested")).toBe("/examples");
+  });
```

Comment:
Good point but actually it's more complicated.

The *parent* from an empty string or `/` should really be null or an error.

This doesn't occur in our code, because we always have at least `/foo` and we never try to get the parent from `/`. So I could keep the method very simple.

But I wonder if I should add extra logic to handle cases without a parent, to prevent us from using it wrongly in the future. What do you think?

In any case, I wouldn't want to add a test for a case that wasn't really intended.